book-1-orientation/chapters/ADONET_INTRO.md

```diff
@@ -9,624 +9,338 @@ This will be an interactive lesson. Your instructor will lead you through follow
+               reader.Close();
+
+               // Return the list of departments who whomever called this method.
+               return departments;
+           }
+       }
+   }
+   ```
+
+   To test this, let's call this method from the `Program.Main` method. Create a new instance of a `DepartmentRepository` and call `GetAllDepartments`. Run the application.
+
+   ```csharp
+   static void Main(string[] args)
+   {
+       DepartmentRepository departmentRepo = new DepartmentRepository();
+
+       Console.WriteLine("Getting All Departments:");
+       Console.WriteLine();
+
+       List<Department> allDepartments = departmentRepo.GetAllDepartments();
+
+       foreach (Department dept in allDepartments)
+       {
+           Console.WriteLine($"{dept.Id} {dept.DeptName}");
+       }
+   }
+   ```
+
+1. Now create another method in `DepartmentRepository` that will get a single department by its Id. The method should accept an `int id` as a paramter and return a single `Department` object.
+
+   ```csharp
+   /// <summary>
+   ///  Returns a single department with the given id.
+   /// </summary>
+   public Department GetDepartmentById(int id)
+   {
+       using (SqlConnection conn = Connection)
+       {
+           conn.Open();
+           using (SqlCommand cmd = conn.CreateCommand())
+           {
+               cmd.CommandText = "SELECT DeptName FROM Department WHERE Id = @id";
+               cmd.Parameters.Add(new SqlParameter("@id", id));
+               SqlDataReader reader = cmd.ExecuteReader();
+
+               Department department = null;
+
+               // If we only expect a single row back from the database, we don't need a while loop.
+               if (reader.Read())
+               {
+                   department = new Department
+                   {
+                       Id = id,
+                       DeptName = reader.GetString(reader.GetOrdinal("DeptName"))
+                   };
+               }
+
+               reader.Close();
+
+               return department;
+           }
+       }
+   }
+   ```
+
+1. Update `Program.Main` to also call this new method
+
+   ```csharp
+   Console.WriteLine("----------------------------");
+   Console.WriteLine("Getting Department with Id 1");
+
+   Department singleDepartment = departmentRepo.GetDepartmentById(1);
+
+   Console.WriteLine($"{singleDepartment.Id} {singleDepartment.DeptName}");
+   ```
+
+1. Now that we've read data from our database, let's look at how we can add new records. Create a new method in the `DepartmentRepository` and name it `AddDepartment`. It should accept a single `Department dept` parameter and also return a Department object.
+
+   ```csharp
+   /// <summary>
+   ///  Add a new department to the database
+   ///   NOTE: This method sends data to the database,
+   ///   it does not get anything from the database, so there is nothing to return.
+   /// </summary>
+   public Department AddDepartment(Department department)
```

Comment:
I'm torn here. Do we want to return the department or just set the ID and rely on mutating the Department object.

Reply:
I like the idea of returning it. EF might not work that way though (IIRC it just mutates it). Is that why you're torn?

Reply:
EF does just update the object. 
I'm concerned that with the code above, the object is being mutated anyway, but if you return an object as well, it _implies_ that it's a new object.

Reply:
Changed